eip-0020.md

```diff
@@ -0,0 +1,196 @@
+formats defined in EIP-0019.
+
+Depending on the bytes size of `ReducedTransaction` it may not fit into 4K limit
+of QR codes. In this case or for whatever reason the dApp may encode the
+following data in the QR code on the payment screen.
+
+```
+ErgoPaySigningRequestUrl
+  - messageType: Byte = 2  // request url
+  - url: String
+```
+
+When the Hot Wallet application parses the QR code bytes it can use first byte
+(the `messageType` field) to learn the message type.
+When `ErgoPaySigningRequestUrl` is parsed, the Hot Wallet application should
+request dApp and obtain the following data.
+
+```
+ErgoPaySigningRequest:
+  - messageType: Byte = 1  // request itself
+  - transaction: ReducedTransaction
+  - description: String
+```
+Once the transaction details obtained, they should be shown on the payment
+screen of the Hot Wallet application.
+
+After signing is performed and the `SignedTransaction` data is obtained, the Hot
+Wallet need to send the following data to the dApp.
+
+```
+ErgoPayTransactionId:
+  - txId: String
+```
+
+After `ErgoPayTransactionId` is sent, only then the dApp can submit the
+SignedTransaction to the blockchain. The order is important, otherwise, if
+transaction is included into blockchain and dApp notification fails the dApp
+will be out of sync with blockchain.
+
+## Implementation in Mobile Wallet
+TODO
+
+## Implementation in dApp
+TODO
+
+## Benefits for dApps
+Ergo Pay provides a fast, easy, and secure way for users to buy goods and
+services in a dApp or on a website. When supported, Ergo Pay can substantially
+increase checkout conversion rates, user loyalty and purchase frequency, and
+reduced checkout time. 
+
+- dApp or website don’t need to handle user's secrets (mnemonic/private keys).
+Instead, once the user has signed the transaction to confirm purchase intent,
+your app or website receives a transaction id to monitor payment status on the
+blockchain.
+- dApp's users don't need to worry about security of their private keys as the
+mobile wallet guarantees they never leave the mobile device
+- ErgoPay EIP is compatible with Cold Wallet EIP, thus users can use Cold Wallet
+devices to sign transaction within ErgoPay signing process.
+- adding Ergo Pay to product detail pages, the cart, checkout page, in payment
+settings, or anywhere else a user can choose ErgoPay as the payment method or
+initiate a purchase.
+- The payment screen can be presented immediately after the user taps the Ergo
+Pay button, without any interim screens or pop-ups except to prompt for
+necessary product details, such as size or quantity.
+- Ergo Pay is simple and universal. It supports all smart contracts and offers
+the flexibility to implement simple to complex dApps.
+
+## Benefits for Mobile Wallets
+
+Any mobile wallet team should consider supporting Ergo Pay in their wallet along
+with basic wallet features. Users can participate in Ergo dApps and mobile
+wallet team can receive service fees from those transactions. 
```

Comment:
yes, in this way, it may be acceptable as long as the user aware of the additional fees.
We can describe this in more details as a second revision of the standard if necessary.